### PR TITLE
Script to run a build against each commit from a failed CI build.

### PR DESCRIPTION
**Pull request by shaahji**
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: examples/test/local/test_bert_cuda_gpu.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ def setup():
     os.chdir(get_example_dir("bert"))
 
 
+@pytest.mark.skip(reason="Disable failing tests")
```
Comment (jambayk) [inline examples/test/local/test_bert_cuda_gpu.py:18]: these tests don't fail in the main ci pipeline so I don't think we should skip them. For the nightly pipelines, once we move to docker runs, we should be able to fix them too.

File: test/unit_test/passes/onnx/test_mnb_to_qdq.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def forward(self, x):
 @pytest.mark.parametrize("use_transpose_op", [True, False])
 @pytest.mark.parametrize("use_int4", [True, False])
 @pytest.mark.parametrize("add_zero_point", [True, False])
-@pytest.mark.parametrize("execution_provider", ["CPUExecutionProvider", "CUDAExecutionProvider"])
+@pytest.mark.parametrize("execution_provider", ["CPUExecutionProvider"])
```
Comment (jambayk) [inline test/unit_test/passes/onnx/test_mnb_to_qdq.py:68]: same for here

File: examples/test/local/test_bert_cuda_gpu.py (hunk 1 above)
Comment (shaahji) [reply examples/test/local/test_bert_cuda_gpu.py:18]: As per my discussion with @devang-ml, the decision is to disable any and all failing tests and get green build. I will send out a list of all disabled tests which will be assigned to developers to fix and enable while keeping all lights green.

The changes in this PR makes all 4 Olive pipelines green.

File: .azure_pipelines/scripts/ClientPatch.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+Related to question in following issue:
+https://github.com/microsoft/azure-devops-python-api/issues/461
+
+"""
+
+import logging
+from typing import Optional, cast
+
+from azure.devops import _models
```
Comment (jambayk) [inline .azure_pipelines/scripts/ClientPatch.py:11]: maybe you can add a requirements.txt file for the dependencies to run these scripts

File: examples/test/local/test_bert_cuda_gpu.py (hunk 1 above)
Comment (jambayk) [reply examples/test/local/test_bert_cuda_gpu.py:18]: I see. thanks for the info.

File: .azure_pipelines/scripts/find_failed_commit.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    access_token = os.getenv("DEVOPS_ACCESS_TOKEN")
+    if not access_token:
+        raise ValueError(
+            "Set DEVOPS_ACCESS_TOKEN in the environment. For instructions on how to get one, visit "
+            "https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows"
+        )
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--project-name", default="PublicPackages", type=str, help="Project name")
+    parser.add_argument("--pipeline-name", default='"Olive-ORT-Nightly', type=str, help="Pipeline name")
```
Comment (jambayk) [inline .azure_pipelines/scripts/find_failed_commit.py:51]: `'"Olive-ORT-Nightly'` has an extra quote. maybe came from linting

File: .azure_pipelines/scripts/find_failed_commit.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    access_token = os.getenv("DEVOPS_ACCESS_TOKEN")
+    if not access_token:
+        raise ValueError(
+            "Set DEVOPS_ACCESS_TOKEN in the environment. For instructions on how to get one, visit "
+            "https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows"
+        )
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--project-name", default="PublicPackages", type=str, help="Project name")
+    parser.add_argument("--pipeline-name", default="Olive-ORT-Nightly", type=str, help="Pipeline name")
```
Comment (jambayk) [inline .azure_pipelines/scripts/find_failed_commit.py:51]: btw, it doesn't seem like pipeline-name is used at all here?

**Review by jambayk — APPROVED** [review]

**Pull request merged**